File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.cpp
class Solution {
public:
    int maxProfit(vector<int>& prices) {
        int profit = 0;
        int price = prices[0u];
        
        for(int i = 1; i < prices.size(); i++){
            if(price > prices[i]){
                price = prices[i];
            }
            else{
                profit = max(profit, prices[i] - price);
            }
        }
        
        return profit;
    }
};

File: 121-best-time-to-buy-and-sell-stock/121-best-time-to-buy-and-sell-stock.java
class Solution {
    public int maxProfit(int[] prices) {
        int local_min = prices[0];
        int result = 0;
        
        for (int i = 1; i < prices.length; i++){
            if (prices[i] < local_min){
                local_min = prices[i];
            }
            else{
                result = Math.max(prices[i] - local_min, result);
            }
        }
        
        return result;
    }
}